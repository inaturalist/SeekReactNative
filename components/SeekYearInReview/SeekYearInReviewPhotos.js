// @flow

import React from "react";
import {
  View,
  Image,
  Pressable
} from "react-native";
import { useNavigation } from "@react-navigation/native";
import type { Node } from "react";

import {
  viewStyles,
  textStyles,
  imageStyles
} from "../../styles/seekYearInReview/seekYearInReview";
import i18n from "../../i18n";
import HorizontalScroll from "../UIComponents/HorizontalScroll";
import StyledText from "../UIComponents/StyledText";
import { SpeciesDetailContext } from "../UserContext";
import { useSeenTaxa, useUserPhoto } from "../../utility/customHooks";

const SeekYearInReviewPhotoItem = ( { observation } ): Node => {
  const seenTaxa = useSeenTaxa( observation?.taxon?.id );
  const userPhoto = useUserPhoto( seenTaxa );

  const { setId } = React.useContext( SpeciesDetailContext );
  const navigation = useNavigation();

  const navToSpecies = () => {
    if ( !observation?.taxon?.id ) {
      return;
    }
    setId( observation.taxon.id );
    navigation.navigate( "Species" );
  };

  return (
    <Pressable
      key={`image${observation.taxon.defaultPhoto.mediumUrl}`}
      style={viewStyles.center}
      onPress={() => navToSpecies()}
    >
      <Image style={imageStyles.image} source={userPhoto} />
      <StyledText style={[textStyles.text, textStyles.caption]}>
        {i18n.t( "seek_year_in_review.observed_on", {
          speciesName:
            observation?.taxon?.preferredCommonName || observation?.taxon?.name,
          date: observation?.date?.toLocaleDateString( i18n.locale )
        } )}
      </StyledText>
    </Pressable>
  );
};

const SeekYearInReviewPhotos = ( { observations } ): Node => {
  const renderPhotos = () =>
    observations.map( ( obs ) => <SeekYearInReviewPhotoItem observation={obs} /> );
  const photoList = renderPhotos();

  return (
    <View>
      {photoList.length === 0 ? null : (
        <HorizontalScroll photoList={photoList} />
      )}
    </View>
  );
};

export default SeekYearInReviewPhotos;
