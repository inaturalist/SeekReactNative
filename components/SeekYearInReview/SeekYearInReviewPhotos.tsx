import React from "react";
import {
  View,
  Image,
  Pressable
} from "react-native";
import { useNavigation } from "@react-navigation/native";

import {
  viewStyles,
  textStyles,
  imageStyles
} from "../../styles/seekYearInReview/seekYearInReview";
import i18n from "../../i18n";
import HorizontalScroll from "../UIComponents/HorizontalScroll";
import StyledText from "../UIComponents/StyledText";
import { useSeenTaxa } from "../../utility/customHooks/useSeenTaxa";
import { useUserPhoto } from "../../utility/customHooks/useUserPhoto";
import { formatDateToDisplayShort } from "../../utility/dateHelpers";
import { setRoute, StoredRoutes } from "../../utility/helpers";
import { useSpeciesDetail } from "../Providers/SpeciesDetailProvider";
import { baseTextStyles } from "../../styles/textStyles";

const SeekYearInReviewPhotoItem = ( { observation, index } ) => {
  const seenTaxa = useSeenTaxa( observation?.taxon?.id );
  const userPhoto = useUserPhoto( seenTaxa );

  const { setId } = useSpeciesDetail();
  const navigation = useNavigation();

  const navToSpecies = () => {
    if ( !observation?.taxon?.id ) {
      return;
    }
    setId( observation.taxon.id );
    setRoute( StoredRoutes.SeekYearInReview );
    navigation.navigate( "Species" );
  };

  return (
    <Pressable
      key={index}
      style={viewStyles.center}
      onPress={() => navToSpecies()}
    >
      {userPhoto?.uri && <Image source={userPhoto} style={imageStyles.image} />}
      <StyledText style={[baseTextStyles.body, textStyles.caption]}>
        {i18n.t( "seek_year_in_review.observed_on", {
          speciesName:
            observation?.taxon?.preferredCommonName || observation?.taxon?.name,
          date: formatDateToDisplayShort( observation?.date )
        } )}
      </StyledText>
    </Pressable>
  );
};

const SeekYearInReviewPhotos = ( { observations } ) => {
  const renderPhotos = () =>
    observations.map( ( obs, index ) => <SeekYearInReviewPhotoItem observation={obs} index={index} /> );
  const photoList = renderPhotos();

  return (
    <View testID="year-in-review-photos-container">
      {photoList.length === 0 ? null : (
        <HorizontalScroll photoList={photoList} />
      )}
    </View>
  );
};

export default SeekYearInReviewPhotos;
