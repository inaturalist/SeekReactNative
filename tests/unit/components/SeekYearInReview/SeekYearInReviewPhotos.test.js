import React from "react";
import { render, screen, fireEvent } from "tests/jest-utils";

import SeekYearInReviewPhotos from "../../../../components/SeekYearInReview/SeekYearInReviewPhotos";

const mockNavigate = {
  navigate: jest.fn()
};

jest.mock( "@react-navigation/native", () => {
  const actualNav = jest.requireActual( "@react-navigation/native" );
  return {
    ...actualNav,
    useNavigation: () => ( {
      ...mockNavigate,
      dispatch: jest.fn()
    } ),
    useRoute: () => ( {} )
  };
} );

const mockObservations = [
  {
    date: new Date( "2022-12-02T10:19:54.000Z" ),
    latitude: 42,
    longitude: 42,
    taxon: {
      ancestorIds: [1, 2, 3],
      defaultPhoto: {
        backupUri: "some_uri",
        lastUpdated: null,
        mediumUrl: "some_medium_url"
      },
      iconicTaxonId: 1,
      id: 4242,
      name: "some_name_1",
      preferredCommonName: "some_common_name_1"
    },
    uuidString: "some_uuid_2"
  },
  {
    date: new Date( "2022-12-01T10:19:54.000Z" ),
    latitude: 42,
    longitude: 42,
    taxon: {
      ancestorIds: [1, 2, 3],
      defaultPhoto: {
        backupUri: "some_uri",
        lastUpdated: null,
        mediumUrl: "some_medium_url"
      },
      iconicTaxonId: 1,
      id: 4242,
      name: "some_name_2",
      preferredCommonName: null
    },
    uuidString: "some_uuid_2"
  },
  {
    date: new Date( "2022-12-03T10:19:54.000Z" ),
    latitude: 42,
    longitude: 42,
    taxon: {
      ancestorIds: [1, 2, 3],
      defaultPhoto: {
        backupUri: "some_uri",
        lastUpdated: null,
        mediumUrl: "some_medium_url"
      },
      iconicTaxonId: 1,
      id: 4242,
      name: "some_name_3",
      preferredCommonName: null
    },
    uuidString: "some_uuid_2"
  }
];

jest.mock( "../../../../utility/customHooks", () => {
  const actual = jest.requireActual( "../../../../utility/customHooks" );
  return {
    ...actual,
    __esModule: true,
    useSeenTaxa: jest.fn( () => mockObservations[0] ),
    useUserPhoto: jest.fn( () => ( {
        uri: "some_uri"
    } ) )
  };
} );

const renderPhotos = () => {
  return render(
    <SeekYearInReviewPhotos observations={mockObservations} />
  );
};

const containerID = "year-in-review-photos-container";

const scrollEventData = {
  nativeEvent: {
    contentOffset: {
      x: 500
    },
    contentSize: {
      // Dimensions of the scrollable content
      height: 500,
      width: 100
    },
    layoutMeasurement: {
      // Dimensions of the device
      height: 100,
      width: 100
    }
  }
};

describe( "SeekYearInReviewPhotos", () => {
  test( "should render correctly", async () => {
    renderPhotos();

    const container = await screen.findByTestId( containerID );
    expect( container ).toBeTruthy();
    // First obs photo should be visible
    const description1 = await screen.findByText( /some_common_name_1/ );
    expect( description1 ).toBeTruthy();
    // Right arrow should be visible
    const rightArrow = await screen.findByTestId( "right-arrow" );
    expect( rightArrow ).toBeTruthy();
    // Second obs photo should not be visible
    const description2 = screen.queryByText( /some_name_3/ );
    expect( description2 ).toBeNull();

    // TODO: errors out on CI because it uses local timezone to display the date string in the photo description
    // expect( screen ).toMatchSnapshot();
  } );

  test( "should show third photo after scroll", async () => {
    renderPhotos();

    // Scroll list to second photo
    const horizontalScroll = await screen.findByTestId( "horizontal-scroll" );
    fireEvent.scroll( horizontalScroll, scrollEventData );

    const description2 = await screen.findAllByText( /some_name_3/ );
    expect( description2 ).toBeTruthy();
  } );

  test( "should navigate to species detail on photo press", async () => {
    renderPhotos();

    const description1 = await screen.findByText( /some_common_name_1/ );
    fireEvent.press( description1 );

    expect( mockNavigate.navigate ).toHaveBeenCalledWith( "Species" );
  } );
} );
