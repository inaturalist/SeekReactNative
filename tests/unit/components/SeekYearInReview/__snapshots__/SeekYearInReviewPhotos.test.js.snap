// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`SeekYearInReviewPhotos should render correctly 1`] = `
<View
  testID="year-in-review-photos-container"
>
  <View
    style={
      {
        "height": 250,
      }
    }
  >
    <View
      style={{}}
    >
      <RCTScrollView
        applyWindowCorrection={[Function]}
        bounces={false}
        canChangeSize={true}
        contentContainerStyle={
          {
            "backgroundColor": "#000000",
            "minHeight": 1,
            "minWidth": 1,
            "paddingBottom": 0,
            "paddingTop": 0,
          }
        }
        contentHeight={900}
        contentWidth={1050}
        data={
          [
            <SeekYearInReviewPhotoItem
              index={0}
              observation={
                {
                  "date": 2022-12-02T10:19:54.000Z,
                  "latitude": 42,
                  "longitude": 42,
                  "taxon": {
                    "ancestorIds": [
                      1,
                      2,
                      3,
                    ],
                    "defaultPhoto": {
                      "backupUri": "some_uri",
                      "lastUpdated": null,
                      "mediumUrl": "some_medium_url",
                    },
                    "iconicTaxonId": 1,
                    "id": 4242,
                    "name": "some_name_1",
                    "preferredCommonName": "some_common_name_1",
                  },
                  "uuidString": "some_uuid_2",
                }
              }
            />,
            <SeekYearInReviewPhotoItem
              index={1}
              observation={
                {
                  "date": 2022-12-01T10:19:54.000Z,
                  "latitude": 42,
                  "longitude": 42,
                  "taxon": {
                    "ancestorIds": [
                      1,
                      2,
                      3,
                    ],
                    "defaultPhoto": {
                      "backupUri": "some_uri",
                      "lastUpdated": null,
                      "mediumUrl": "some_medium_url",
                    },
                    "iconicTaxonId": 1,
                    "id": 4242,
                    "name": "some_name_2",
                    "preferredCommonName": null,
                  },
                  "uuidString": "some_uuid_2",
                }
              }
            />,
            <SeekYearInReviewPhotoItem
              index={2}
              observation={
                {
                  "date": 2022-12-03T10:19:54.000Z,
                  "latitude": 42,
                  "longitude": 42,
                  "taxon": {
                    "ancestorIds": [
                      1,
                      2,
                      3,
                    ],
                    "defaultPhoto": {
                      "backupUri": "some_uri",
                      "lastUpdated": null,
                      "mediumUrl": "some_medium_url",
                    },
                    "iconicTaxonId": 1,
                    "id": 4242,
                    "name": "some_name_3",
                    "preferredCommonName": null,
                  },
                  "uuidString": "some_uuid_2",
                }
              }
            />,
          ]
        }
        dataProvider={
          DataProvider {
            "_data": [
              <SeekYearInReviewPhotoItem
                index={0}
                observation={
                  {
                    "date": 2022-12-02T10:19:54.000Z,
                    "latitude": 42,
                    "longitude": 42,
                    "taxon": {
                      "ancestorIds": [
                        1,
                        2,
                        3,
                      ],
                      "defaultPhoto": {
                        "backupUri": "some_uri",
                        "lastUpdated": null,
                        "mediumUrl": "some_medium_url",
                      },
                      "iconicTaxonId": 1,
                      "id": 4242,
                      "name": "some_name_1",
                      "preferredCommonName": "some_common_name_1",
                    },
                    "uuidString": "some_uuid_2",
                  }
                }
              />,
              <SeekYearInReviewPhotoItem
                index={1}
                observation={
                  {
                    "date": 2022-12-01T10:19:54.000Z,
                    "latitude": 42,
                    "longitude": 42,
                    "taxon": {
                      "ancestorIds": [
                        1,
                        2,
                        3,
                      ],
                      "defaultPhoto": {
                        "backupUri": "some_uri",
                        "lastUpdated": null,
                        "mediumUrl": "some_medium_url",
                      },
                      "iconicTaxonId": 1,
                      "id": 4242,
                      "name": "some_name_2",
                      "preferredCommonName": null,
                    },
                    "uuidString": "some_uuid_2",
                  }
                }
              />,
              <SeekYearInReviewPhotoItem
                index={2}
                observation={
                  {
                    "date": 2022-12-03T10:19:54.000Z,
                    "latitude": 42,
                    "longitude": 42,
                    "taxon": {
                      "ancestorIds": [
                        1,
                        2,
                        3,
                      ],
                      "defaultPhoto": {
                        "backupUri": "some_uri",
                        "lastUpdated": null,
                        "mediumUrl": "some_medium_url",
                      },
                      "iconicTaxonId": 1,
                      "id": 4242,
                      "name": "some_name_3",
                      "preferredCommonName": null,
                    },
                    "uuidString": "some_uuid_2",
                  }
                }
              />,
            ],
            "_firstIndexToProcess": 0,
            "_hasStableIds": false,
            "_requiresDataChangeHandling": false,
            "_size": 3,
            "getStableId": [Function],
            "rowHasChanged": [Function],
          }
        }
        disableRecycling={false}
        estimatedItemSize={350}
        extendedState={{}}
        externalScrollView={[Function]}
        finalRenderAheadOffset={250}
        forceNonDeterministicRendering={true}
        getItemLayout={[Function]}
        horizontal={true}
        indicatorStyle="white"
        initialNumToRender={1}
        initialOffset={0}
        initialRenderIndex={0}
        isHorizontal={true}
        layoutProvider={
          GridLayoutProviderWithProps {
            "_acceptableRelayoutDelta": 1,
            "_getHeightOrWidth": [Function],
            "_getLayoutTypeForIndex": [Function],
            "_getSpan": [Function],
            "_hasExpired": false,
            "_isHorizontal": true,
            "_lastLayoutManager": GridLayoutManager {
              "_acceptableRelayoutDelta": 1,
              "_getSpan": [Function],
              "_isGridHorizontal": true,
              "_isHorizontal": true,
              "_layoutProvider": [Circular],
              "_layouts": [
                {
                  "height": 900,
                  "type": 0,
                  "width": 350,
                  "x": 0,
                  "y": 0,
                },
                {
                  "height": 900,
                  "type": 0,
                  "width": 350,
                  "x": 350,
                  "y": 0,
                },
                {
                  "height": 900,
                  "type": 0,
                  "width": 350,
                  "x": 700,
                  "y": 0,
                },
              ],
              "_maxSpan": 1,
              "_renderWindowSize": {
                "height": 900,
                "width": 400,
              },
              "_totalHeight": 900,
              "_totalWidth": 1050,
              "_window": {
                "height": 900,
                "width": 400,
              },
            },
            "_maxSpan": 1,
            "_renderWindowSize": {
              "height": 900,
              "width": 400,
            },
            "_setLayoutForType": [Function],
            "_tempDim": {
              "height": 0,
              "width": 0,
            },
            "averageWindow": AverageWindow {
              "currentAverage": 350,
              "currentCount": 1,
              "inputValues": [
                350,
                ,
                ,
                ,
                ,
                ,
              ],
              "nextIndex": 1,
            },
            "defaultEstimatedItemSize": 100,
            "layoutObject": {
              "size": undefined,
              "span": undefined,
            },
            "props": {
              "bounces": false,
              "contentContainerStyle": {
                "backgroundColor": "#000000",
              },
              "data": [
                <SeekYearInReviewPhotoItem
                  index={0}
                  observation={
                    {
                      "date": 2022-12-02T10:19:54.000Z,
                      "latitude": 42,
                      "longitude": 42,
                      "taxon": {
                        "ancestorIds": [
                          1,
                          2,
                          3,
                        ],
                        "defaultPhoto": {
                          "backupUri": "some_uri",
                          "lastUpdated": null,
                          "mediumUrl": "some_medium_url",
                        },
                        "iconicTaxonId": 1,
                        "id": 4242,
                        "name": "some_name_1",
                        "preferredCommonName": "some_common_name_1",
                      },
                      "uuidString": "some_uuid_2",
                    }
                  }
                />,
                <SeekYearInReviewPhotoItem
                  index={1}
                  observation={
                    {
                      "date": 2022-12-01T10:19:54.000Z,
                      "latitude": 42,
                      "longitude": 42,
                      "taxon": {
                        "ancestorIds": [
                          1,
                          2,
                          3,
                        ],
                        "defaultPhoto": {
                          "backupUri": "some_uri",
                          "lastUpdated": null,
                          "mediumUrl": "some_medium_url",
                        },
                        "iconicTaxonId": 1,
                        "id": 4242,
                        "name": "some_name_2",
                        "preferredCommonName": null,
                      },
                      "uuidString": "some_uuid_2",
                    }
                  }
                />,
                <SeekYearInReviewPhotoItem
                  index={2}
                  observation={
                    {
                      "date": 2022-12-03T10:19:54.000Z,
                      "latitude": 42,
                      "longitude": 42,
                      "taxon": {
                        "ancestorIds": [
                          1,
                          2,
                          3,
                        ],
                        "defaultPhoto": {
                          "backupUri": "some_uri",
                          "lastUpdated": null,
                          "mediumUrl": "some_medium_url",
                        },
                        "iconicTaxonId": 1,
                        "id": 4242,
                        "name": "some_name_3",
                        "preferredCommonName": null,
                      },
                      "uuidString": "some_uuid_2",
                    }
                  }
                />,
              ],
              "estimatedItemSize": 350,
              "getItemLayout": [Function],
              "horizontal": true,
              "indicatorStyle": "white",
              "initialNumToRender": 1,
              "numColumns": 1,
              "onViewableItemsChanged": [Function],
              "pagingEnabled": true,
              "renderItem": [Function],
              "showsHorizontalScrollIndicator": false,
              "testID": "horizontal-scroll",
              "viewabilityConfig": {
                "viewAreaCoveragePercentThreshold": 95,
                "waitForInteraction": true,
              },
            },
            "renderWindowInsets": {
              "height": 0,
              "width": 0,
            },
            "shouldRefreshWithAnchoring": true,
          }
        }
        maxRenderAhead={750}
        numColumns={1}
        onEndReached={[Function]}
        onEndReachedThreshold={0}
        onEndReachedThresholdRelative={0}
        onItemLayout={[Function]}
        onLayout={[Function]}
        onScroll={[Function]}
        onScrollBeginDrag={[Function]}
        onSizeChanged={[Function]}
        onViewableItemsChanged={[Function]}
        onVisibleIndicesChanged={[Function]}
        pagingEnabled={true}
        removeClippedSubviews={false}
        renderAheadOffset={0}
        renderAheadStep={250}
        renderContentContainer={[Function]}
        renderItem={[Function]}
        renderItemContainer={[Function]}
        rowRenderer={[Function]}
        scrollEventThrottle={16}
        scrollThrottle={16}
        scrollViewProps={
          {
            "contentContainerStyle": {
              "backgroundColor": "#000000",
              "minHeight": 1,
              "minWidth": 1,
              "paddingBottom": 0,
              "paddingTop": 0,
            },
            "onLayout": [Function],
            "onScrollBeginDrag": [Function],
            "refreshControl": undefined,
            "style": {
              "minHeight": 1,
              "minWidth": 1,
            },
          }
        }
        showsHorizontalScrollIndicator={false}
        style={
          {
            "minHeight": 1,
            "minWidth": 1,
          }
        }
        suppressBoundedSizeException={true}
        testID="horizontal-scroll"
        viewabilityConfig={
          {
            "viewAreaCoveragePercentThreshold": 95,
            "waitForInteraction": true,
          }
        }
        windowCorrectionConfig={
          {
            "applyToInitialOffset": true,
            "applyToItemScroll": true,
            "value": {
              "endCorrection": 0,
              "startCorrection": 0,
              "windowShift": -0,
            },
          }
        }
      >
        <View>
          <View
            style={
              {
                "flexDirection": "row",
              }
            }
          >
            <View
              style={
                {
                  "paddingLeft": 0,
                  "paddingTop": 0,
                }
              }
            />
            <View
              style={
                [
                  undefined,
                  undefined,
                ]
              }
            />
            <AutoLayoutView
              enableInstrumentation={false}
              horizontal={true}
              onBlankAreaEvent={[Function]}
              onLayout={[Function]}
              renderAheadOffset={0}
              scrollOffset={0}
              style={
                {
                  "height": 900,
                  "width": 1050,
                }
              }
              windowSize={400}
            >
              <CellContainer
                index={0}
                onLayout={[Function]}
                style={
                  {
                    "alignItems": "stretch",
                    "flexDirection": "row",
                    "height": 900,
                    "left": 0,
                    "position": "absolute",
                    "top": 0,
                  }
                }
              >
                <View
                  style={
                    {
                      "flexDirection": "column",
                    }
                  }
                >
                  <View
                    accessible={true}
                    collapsable={false}
                    focusable={true}
                    onBlur={[Function]}
                    onClick={[Function]}
                    onFocus={[Function]}
                    onResponderGrant={[Function]}
                    onResponderMove={[Function]}
                    onResponderRelease={[Function]}
                    onResponderTerminate={[Function]}
                    onResponderTerminationRequest={[Function]}
                    onStartShouldSetResponder={[Function]}
                    style={
                      {
                        "alignItems": "center",
                        "justifyContent": "center",
                      }
                    }
                  >
                    <Image
                      source={
                        {
                          "uri": "some_uri",
                        }
                      }
                      style={
                        {
                          "height": 186,
                          "resizeMode": "cover",
                          "width": 750,
                        }
                      }
                    />
                    <Text
                      style={
                        [
                          {
                            "textAlign": "left",
                          },
                          [
                            {
                              "color": "#000000",
                              "fontFamily": "Whitney-Book",
                              "fontSize": 16,
                              "lineHeight": 21,
                              "textAlign": "center",
                            },
                            {
                              "marginTop": 16,
                              "maxWidth": 245,
                              "textAlign": "center",
                            },
                          ],
                        ]
                      }
                    >
                      some_common_name_1 observed on Dec 2, 2022, 11:19 AM
                    </Text>
                  </View>
                </View>
              </CellContainer>
              <CellContainer
                index={1}
                onLayout={[Function]}
                style={
                  {
                    "alignItems": "stretch",
                    "flexDirection": "row",
                    "height": 900,
                    "left": 350,
                    "position": "absolute",
                    "top": 0,
                  }
                }
              >
                <View
                  style={
                    {
                      "flexDirection": "column",
                    }
                  }
                >
                  <View
                    accessible={true}
                    collapsable={false}
                    focusable={true}
                    onBlur={[Function]}
                    onClick={[Function]}
                    onFocus={[Function]}
                    onResponderGrant={[Function]}
                    onResponderMove={[Function]}
                    onResponderRelease={[Function]}
                    onResponderTerminate={[Function]}
                    onResponderTerminationRequest={[Function]}
                    onStartShouldSetResponder={[Function]}
                    style={
                      {
                        "alignItems": "center",
                        "justifyContent": "center",
                      }
                    }
                  >
                    <Image
                      source={
                        {
                          "uri": "some_uri",
                        }
                      }
                      style={
                        {
                          "height": 186,
                          "resizeMode": "cover",
                          "width": 750,
                        }
                      }
                    />
                    <Text
                      style={
                        [
                          {
                            "textAlign": "left",
                          },
                          [
                            {
                              "color": "#000000",
                              "fontFamily": "Whitney-Book",
                              "fontSize": 16,
                              "lineHeight": 21,
                              "textAlign": "center",
                            },
                            {
                              "marginTop": 16,
                              "maxWidth": 245,
                              "textAlign": "center",
                            },
                          ],
                        ]
                      }
                    >
                      some_name_2 observed on Dec 1, 2022, 11:19 AM
                    </Text>
                  </View>
                </View>
              </CellContainer>
            </AutoLayoutView>
            <CellContainer
              index={-1}
              style={
                [
                  undefined,
                  undefined,
                ]
              }
            />
            <View
              style={
                {
                  "paddingBottom": 0,
                  "paddingRight": 0,
                }
              }
            />
            <View
              pointerEvents="none"
              style={
                {
                  "opacity": 0,
                }
              }
            >
              <View
                accessible={true}
                collapsable={false}
                focusable={true}
                onBlur={[Function]}
                onClick={[Function]}
                onFocus={[Function]}
                onResponderGrant={[Function]}
                onResponderMove={[Function]}
                onResponderRelease={[Function]}
                onResponderTerminate={[Function]}
                onResponderTerminationRequest={[Function]}
                onStartShouldSetResponder={[Function]}
                style={
                  {
                    "alignItems": "center",
                    "justifyContent": "center",
                  }
                }
              >
                <Image
                  source={
                    {
                      "uri": "some_uri",
                    }
                  }
                  style={
                    {
                      "height": 186,
                      "resizeMode": "cover",
                      "width": 750,
                    }
                  }
                />
                <Text
                  style={
                    [
                      {
                        "textAlign": "left",
                      },
                      [
                        {
                          "color": "#000000",
                          "fontFamily": "Whitney-Book",
                          "fontSize": 16,
                          "lineHeight": 21,
                          "textAlign": "center",
                        },
                        {
                          "marginTop": 16,
                          "maxWidth": 245,
                          "textAlign": "center",
                        },
                      ],
                    ]
                  }
                >
                  some_name_2 observed on Dec 1, 2022, 11:19 AM
                </Text>
              </View>
            </View>
          </View>
        </View>
      </RCTScrollView>
    </View>
    <View
      accessibilityLabel="Scroll right"
      accessible={true}
      collapsable={false}
      focusable={true}
      onClick={[Function]}
      onResponderGrant={[Function]}
      onResponderMove={[Function]}
      onResponderRelease={[Function]}
      onResponderTerminate={[Function]}
      onResponderTerminationRequest={[Function]}
      onStartShouldSetResponder={[Function]}
      style={
        {
          "opacity": 1,
          "paddingLeft": 20,
          "paddingRight": 5,
          "paddingVertical": 20,
          "position": "absolute",
          "right": 0,
          "top": 1010,
          "zIndex": 1,
        }
      }
      testID="right-arrow"
    >
      <Image
        source={
          {
            "testUri": "../../../assets/icons/img-swipearrow-right.webp",
          }
        }
        style={
          {
            "transform": [
              {
                "rotate": "0deg",
              },
            ],
          }
        }
      />
    </View>
  </View>
</View>
`;
